### PR TITLE
update to eap 2020.1, update of gradle plugin required as well

### LINKED ISSUE #53: plugin doesn't work with EAP 2021.1 / "Please ensure JavaScript is enabled."
Hi folks, 

Trying this plugin in latest version of WebStorm (EAP) but when I try to open a diagram it says:

```
Page could not be loaded. Please try refreshing.
Please ensure JavaScript is enabled.
```

I cannot find any information about how to get this to work Am I missing something?

Cheers, 
Danny

### PR DESCRIPTION
**Pull request by ahus1**
This updates the IntelliJ platform to 2021.1 EAP. The problem can be replicated using this setup.

closes #53

### CODE REVIEW

**Comment by ahus1** [comment]
I don't know how to fix, this but I managed to reproduce the issue. Some insight in the JavaScript parts of diagrams.net would be required. // cc: @rdmueller + @hediet 

The error message as displayed in the browser:

image

The error message as displayed in the browser's console (right-click on preview to open the Open the DevTools)

image

It seems that due to the new schema "drawio-plugin" the browser doesn't allow access to cookies any more. I searched the Internet, but didn't find a solution for that. 

I tried to change "drawio-plugin://main..." to "https://main..." and changed also in registerSchemeHandlerFactory the "drawio-plugin" to "https" and "*" to "main". The the JavaScript initializes, and show the following dialog, and it seems that it doesn't load the diagram: 

image

Any help would be appreciated. The branch already contains the changes regarding IntelliJ EAP, but doesn't include the changes about the schema (as I wasn't sure if this is a path to explore). Please let me know if I should push these changes as well.

**Comment by hediet** [comment]
> I tried to change "drawio-plugin://main..." to "https://main..." and changed also in registerSchemeHandlerFactory the "drawio-plugin" to "https" and "*" to "main". The the JavaScript initializes, and show the following dialog, and it seems that it doesn't load the diagram

I don't think overriding the default http/s handler is a good idea.

[This line should avoid this issue](https://github.com/docToolchain/diagrams.net-intellij-plugin/blob/main/src/webview/index.html#L95):
```ts
Object.defineProperty(document, "cookie", { value: "" });
```
I don't know why it does not work anymore.

**Comment by ahus1** [comment]
I debugged and found that at least locally, maybe even in the latest version of the plugin the index.html custom file was not used, instead the one from the submodule was used. Maybe it depended on file timestamps? Anyway, explicitly excluding the submodule one solved it for me.

Once the right index.html file was used, the cookie issue disappeared.

I then received an error that due to CORS policy Cross origin requests would not be allowed for the new protocol scheme. Once I changed the protocol, it started to work again without issues, see 2dd2f91bec0ff89a804e070cf5fc5f7dd382b65d

This is now ready for review. If no-one has a different solution to this, I'd like to publish this to support the latest 2021.1 EAP version (final 2021.1 should be out soon I suppose).

File: src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/BaseDrawioWebView.kt [hunk 1]
```diff
@@ -28,7 +28,7 @@ abstract class BaseDrawioWebView(val lifetime: Lifetime, var uiTheme: String) {
             didRegisterSchemeHandler = true
 
             CefApp.getInstance().registerSchemeHandlerFactory(
-                    "drawio-plugin", "*",
+                    "https", "drawio-plugin",
```
Comment (hediet) [inline src/main/kotlin/de/docs_as_co/intellij/plugin/drawio/editor/BaseDrawioWebView.kt:31]: Can you add a comment here why `drawio-plugin://` does not work?

**Review by hediet — APPROVED** [review]
Please add a comment that explains why `drawio-plugin://` cannot be used. Looks good otherwise!

**Comment by ahus1** [comment]
Thanks for the quick review, PR updated, merging to main.

**Pull request merged**